File: app/src/main/java/awais/instagrabber/repositories/responses/Notification.java
package awais.instagrabber.repositories.responses;

import awais.instagrabber.models.enums.NotificationType;

public class Notification {
    private final NotificationArgs args;
    private final String storyType;
    private final String pk;

    public Notification(final NotificationArgs args,
                        final String storyType,
                        final String pk) {
        this.args = args;
        this.storyType = storyType;
        this.pk = pk;
    }

    public NotificationArgs getArgs() {
        return args;
    }

    public NotificationType getType() {
        return NotificationType.valueOfType(storyType);
    }

    public String getPk() {
        return pk;
    }
}

File: app/src/main/java/awais/instagrabber/repositories/responses/notification/Notification.kt
package awais.instagrabber.repositories.responses.notification

import awais.instagrabber.models.enums.NotificationType
import awais.instagrabber.models.enums.NotificationType.Companion.valueOfType

class Notification(val args: NotificationArgs,
                   private val storyType: Int,
                   val pk: String) {
    val type: NotificationType?
        get() = valueOfType(storyType)
}